### ISSUE TITLE
Select All and Deselect All buttons for Tree

### ISSUE THREAD

**Issue by hiteshc2** [primefaces/primeng#3307]
I'm submitting a ..
[ ] bug report => Search github for a similar issue or PR before submitting
[x] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35

Hi,

It would be helpful if tree had "select all" and "deselect all" buttons enabling programatic selections and deselection of nodes.

**Comment by vscodeguru** [comment]
#3285 I am having a same issue

**Comment by mnhock** [comment]
It would be great if this works like the multiselect from the DataTable (Checkbox in the header).

**Comment by lmsv-mx123** [comment]
Agreed is something good to have. With new PrimeNG 12 on roadmap, if willing to contribute should hold off until it is released?

**Comment by lmsv-mx123** [comment]
Hi @mertsincan,

Thank you for your comments into the issue. I have checked on the documentation for latest version and examined through the latest code and the feature request still is needed. Will greatly appreciated it if can be included in the roadmap.

Best regards

**Comment by lmsv-mx123** [comment]
@hiteshc2 @mertsincan from last comments, could the issue be reopened as I'm not the author of it or have the necessary rights. 

Thank you

**Comment by MohamedTaha2023** [comment]
Yes we need select all and deselect all

**Comment by christian-supy-io** [comment]
its been 9 years....
